snapcraft/explanation/parts-lifecycle.md

```diff
@@ -0,0 +1,90 @@
+
+To access the part environment at any stage, add the `--shell` argument. For example, `snapcraft prime --shell` will run up to the *prime* step and open a shell. See [Iterating over a build](/t/iterating-over-a-build/12143) for more details.
+
+<h3 id='heading--steps-dependencies'>Steps dependencies<sup><a href=#heading--steps-dependencies>⚓</a></sup></h3>
+
+Each lifecycle step depends on the completion of the previous step for that part, so to reach a desired step, all prior steps need to have successfully run. By default, `snapcraft` runs the same lifecycle step of all parts before moving to the next step. However, you can change this behavior using the `after` keyword in the definition of a part in `snapcraft.yaml`. This creates a dependency chain from one part to another.
+
+```yaml
+ grv:
+    plugin: go
+    go-channel: 1.11/stable
+    after:
+      - libgit2
+```
+
+In the above example, the part named `grv` will be built after the part named `libgit2` has been successfully built _and_ staged.
+
+<h3 id='heading--overriding-steps'>Overriding a step<sup><a href=#heading--overriding-steps>⚓</a></sup></h3>
+
+Each plugin defines the default actions that happen during a step. This behavior can be changed in two ways.
+
+- By using `override-<step-name>` in `snapcraft.yaml`. See [Overriding steps](/t/scriptlets/4892) for more details.
+- By using a local plugin.  This can inherit the parent plugin or scaffolding from the original. See [Local plugins](/t/writing-local-plugins/5125) for more details.
+
+See [Parts environment variables](/t/parts-environment-variables/12271) for a list of part-specific environment variables that can be accessed to help build a part.
+
+<h3 id='heading--parts-directories'>Parts directories<sup><a href=#heading--parts-directories>⚓</a></sup></h3>
+
+When running through its build steps, a part will use different working directories. These closely follow the step names for the lifecycle.
+
+The directories are specified by their environment variables:
+- **Snapcraft 7+**: environment variables start with `CRAFT_`
+- **Snapcraft 6, and earlier**: variables start with `SNAPCRAFT_` but are otherwise identical
+
+See [Parts environment variables](/t/parts-environment-variables/12271) for more details.
+
+| Environment variable | Directory | Purpose |
+|--|--|--|
+| `CRAFT_PART_SRC` | **`parts/<part-name>/src`** | the location of the source during the *pull* step |
+| `CRAFT_PART_BUILD` | **`parts/<part-name>/build`** | the working directory during the *build* step |
+| `CRAFT_PART_INSTALL`| **`parts/<part-name>/install`** | contains the results of the *build* step and the stage packages. |
+| `CRAFT_STAGE` | **`stage`** | shared by all parts, this directory contains the development libraries, headers, and other components (e.g.; pkgconfig files) that need to be accessible from other parts |
```

Comment:
"This directory *can* contain" -- it isn't necessary that they do so (depends on the plugin, `stage:`, the build script in the source itself, actions in `override-build`, etc.)
Likewise it *definitely* contains the contents of each part's `CRAFT_PART_INSTALL` sans any likewise shenanigans as above.